enterprise/frontend/src/embedding-sdk/cli/utils/add-database-connection.ts

```diff
@@ -0,0 +1,62 @@
+import { propagateErrorResponse } from "./propagate-error-response";
+
+interface Options {
+  name: string;
+  engine: string;
+  connection: Record<string, string | boolean | number>;
+
+  cookie: string;
+  instanceUrl: string;
+}
+
+export async function addDatabaseConnection(options: Options) {
+  const { instanceUrl, name, engine, connection, cookie } = options;
+
+  const url = `${instanceUrl}/api/database`;
+  const { host } = connection;
+
+  if (connection.ssl) {
+    connection["ssl-mode"] = "require";
+  }
+
+  if (engine === "bigquery-cloud-sdk") {
+    connection["dataset-filters-type"] = "all";
+  }
+
+  // If the host points to localhost, we use docker's
+  // internal DNS to connect to the host database.
+  if (!host || host === "localhost" || host === "127.0.0.1") {
+    connection["host"] = "host.docker.internal";
+  }
```

Comment:
Because Metabase is running within a Docker container, we need to re-map their localhost to `host.docker.internal`